pkg/submitter/submitter.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package submitter
+
+import (
+	"fmt"
+	"sort"
+
+	"sqlflow.org/sqlflow/pkg/database"
+	"sqlflow.org/sqlflow/pkg/pipe"
+	"sqlflow.org/sqlflow/pkg/sql/ir"
+
+	pb "sqlflow.org/sqlflow/pkg/proto"
+)
+
+var submitterRegistry = make(map[string]Submitter)
+
+// Submitter submites the Python program by code generator: `pkg/sql/codegen`
+type Submitter interface {
+	ir.Executor
```

Comment:
Should we move `ir.Executor` out from the package `ir` into `executor`? An IR is not executable, only the submitter is executable.  Because only submitters are executable, we don't need the long name `SubmitterExecutor`, but `Executor` is good enough.

Also, should we not to create package `submitter` and close this PR, but open another PR to create an `executor` package? In that way, we could put this `SubmitterExecutor` (or `Executor`) and `ir.Executor` into the same pacakge.

Reply:
Good idea, maybe we can move to `step` package, and using `step.Context` to setup an `Executor`:

``` golang
package step
type Executor interface {
    setup(ctx *Context)
    ExecuteTrain(ts *ir.TrainStmt)
    ExecutePredict(ts *ir.PredictStmt)
    ExecuteExplain(es *ir.ExplainStmt)
    GetTrainStmtFromModel() bool
}
```

Reply:
Yes. BTW, an interface doesn't contain unexported methods like `setup`. We either remove it or rename it into `Setup`.

Reply:
would rename it to `Setup`.